dotnet/samples/GettingStartedWithVectorStores/Step1_Ingest_Data.cs

```diff
@@ -0,0 +1,111 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using Microsoft.Extensions.VectorData;
+using Microsoft.SemanticKernel.Connectors.InMemory;
+using Microsoft.SemanticKernel.Embeddings;
+
+namespace GettingStartedWithVectorStores;
+
+/// <summary>
+/// Example showing how to generate embeddings and ingest data into an in-memory vector store.
+/// </summary>
+public class Step1_Ingest_Data(ITestOutputHelper output, VectorStoresFixture fixture) : BaseTest(output), IClassFixture<VectorStoresFixture>
+{
+    /// <summary>
+    /// Example showing how to ingest data into an in-memory vector store.
+    /// </summary>
+    [Fact]
+    public async Task IngestDataIntoInMemoryVectorStoreAsync()
+    {
+        // Construct the vector store and get the collection.
+        var vectorStore = new InMemoryVectorStore();
+        var collection = vectorStore.GetCollection<string, Glossary>("skglossary");
+
+        // Ingest data into the collection.
+        await IngestDataIntoVectorStoreAsync(collection, fixture.TextEmbeddingGenerationService);
+
+        // Retrieve an item from the collection and write it to the console.
+        var skDefinition = await collection.GetAsync("4");
+        Console.WriteLine(skDefinition!.Definition);
+    }
+
+    /// <summary>
+    /// Ingest data into the given collection.
+    /// </summary>
+    /// <param name="collection">The collection to ingest data into.</param>
+    /// <param name="textEmbeddingGenerationService">The service to use for generating embeddings.</param>
+    internal static async Task IngestDataIntoVectorStoreAsync(
+        IVectorStoreRecordCollection<string, Glossary> collection,
+        ITextEmbeddingGenerationService textEmbeddingGenerationService)
+    {
+        // Create the collection if it doesn't exist.
+        await collection.CreateCollectionIfNotExistsAsync();
+
+        // Create glossary entries and generate embeddings for them.
+        var glossaryEntries = CreateGlossaryEntries().ToList();
+        var tasks = glossaryEntries.Select(entry => Task.Run(async () =>
+        {
+            entry.DefinitionEmbedding = await textEmbeddingGenerationService.GenerateEmbeddingAsync(entry.Definition);
+        }));
+        await Task.WhenAll(tasks);
+
+        // Upsert the glossary entries into the collection and return their keys.
+        var upsertedKeysTasks = glossaryEntries.Select(x => collection.UpsertAsync(x));
+        var upsertedKeys = await Task.WhenAll(upsertedKeysTasks);
```

Comment:
Should this function return the `upsertedKeys`, seems odd to assign the variable and then not use it